### PR TITLE
Some code cleanup/bug fixes

### PR DESCRIPTION
**Pull request by RappyTV**
### Changes
+ [`ef4b281`](https://github.com/fldardenne/DiscordSINFBot/commit/ef4b281d5dc64f2795a4eecb167685c3eae9f8bd) - Fix of #61, event cleanups, moved interactionCreate into own file
+ [`4d679a4`](https://github.com/fldardenne/DiscordSINFBot/commit/4d679a427a8c89e3a60956d45d774db5ff5c0891) - The commands will now be loaded on startup
+ [`4d9a030`](https://github.com/fldardenne/DiscordSINFBot/commit/4d9a03052070e55d5efa1d71d75dd2ef6eb687e3)/[`6d0962d`](https://github.com/fldardenne/DiscordSINFBot/commit/6d0962d3fbd2043f3ba1a89aeefcf78a8ca7929a) - Code cleanup
+ [`a6f7dd1`](https://github.com/fldardenne/DiscordSINFBot/commit/a6f7dd12c514539a225829dbcaaaabe8cc243b4e) - Fixed some markdown in the readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,16 +2,18 @@
 
 ## How to install ?
 Clone the project and install the dependencies in the `src` folder by running
-```npm install```
+```
```
Comment (obiwac) [inline README.md:5]: While we're at it, we can make this "```console".

File: README.md [hunk 2]
```diff
@@ -2,16 +2,18 @@
...
 
 Create the `src/.env` file, paste in the content of `.env.example` by modifying the variables appropriately.
 To create a bot and get the token, go to https://discord.com/developers/applications.
 
-To "upload" the commands to the Discord helper, run
-```node deploy-commands.js```
-
 To run the bot, execute
-```node bot.js```
+```
```
Comment (obiwac) [inline README.md:13]: ditto

File: README.md [hunk 3]
```diff
@@ -2,16 +2,18 @@
...
+node bot.js
+```
+> The commands will be registered automatically
```
Comment (obiwac) [inline README.md:16]: This doesn't need to be indented and is missing a full stop.

File: src/bot.js [hunk 4]
```diff
@@ -12,57 +12,21 @@ const client = new Client({
...
-
-	else {
-		client.on(event.name, (...args) => event.execute(client, ...args))
-	}
+	if(event.once) client.once(event.name, (...args) => event.execute(client, ...args));
+	else client.on(event.name, (...args) => event.execute(client, ...args));
 }
 
-client.login(process.env.DISCORD_TOKEN)
+client.login(process.env.DISCORD_TOKEN);
```
Comment (obiwac) [inline src/bot.js:32]: Missing terminating newline.

File: src/events/guildMemberAdd.js [hunk 5]
```diff
@@ -1,20 +1,24 @@
...
+     */
 
-N'oublie pas de lire les règles
- `
+    async execute(client, member) {
+        /**********************
+            \n = New line
+            \n\n = Empty line between two lines of text
+        ***********************/
+        const welcomeMessage = `Bienvenue sur le Discord SINF\nCe serveur a pour but de réunir tous les étudiants en informatique en un seul serveur Discord, l'accent est mis sur le partage et l'entraide entre étudiants.\nVous avez ici le droit de discuter, poser vos questions, donner votre avis sur un cours, partager vos tuyaux/synthèses/découvertes, recruter/chercher des personnes pour un projet (non limité aux cours), jouer, ...\nNous organisons également régulièrement des petits événements: le meme contest, distribution de stickers, concours de celui qui a le plus beau chat/setup/...\n\nPour l'instant, tu as seulement accès aux channels communautaires.\n\nUtilise la commande /register n'importe où sur le serveur pour t'inscrire aux channels de cours.\n\nN'oublie pas de lire les règles`;
```
Comment (obiwac) [inline src/events/guildMemberAdd.js:20]:
```suggestion
        const welcomeMessage = `Bienvenue sur le Discord SINF\nCe serveur a pour but de réunir tous les étudiants en informatique en un seul serveur Discord. L'accent est mis sur le partage et l'entraide entre étudiants.\nVous avez ici le droit de discuter, poser vos questions, donner votre avis sur un cours, partager vos tuyaux/synthèses/découvertes, recruter/chercher des personnes pour un projet (non limité aux cours), jouer, ...\nNous organisons également régulièrement des petits événements : le meme contest, distribution de stickers, concours de celui qui a le plus beau chat/setup/...\n\nPour l'instant, tu n'as seulement accès aux channels communautaires.\n\nUtilise la commande /register n'importe où sur le serveur pour t'inscrire aux channels de cours.\n\nN'oublie pas de lire les règles.`;
```

File: src/events/interactionCreate.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    
+        try {
+            console.log(command);
+            await command.execute(client, interaction);
+        } catch(err) {
+            console.error(err);
+            return interaction.reply({ content: 'There was an error while executing this command!', ephemeral: true });
+        }
+    },
+}
```
Comment (obiwac) [inline src/events/interactionCreate.js:29]: Missing newline.

File: src/events/ready.js [hunk 7]
```diff
@@ -1,7 +1,39 @@
...
+        const command = require(`../commands/${file}`);
+        commands.push(command.data.toJSON());
+    }
+
+    rest.put(Routes.applicationGuildCommands(process.env.CLIENT_ID, process.env.GUILD_ID), {
+        body: commands
+    })
+    .then(() => console.log('Successfully registered application commands.'))
+    .catch(console.error);
+}
```
Comment (obiwac) [inline src/events/ready.js:39]: Missing newline.

**Review by obiwac — CHANGES_REQUESTED** [review]
I'll have to try running it, but otherwise all seems good, thank you!

File: README.md (hunk 1 above)
Comment (RappyTV) [reply README.md:5]: wdym by this?

Comment (obiwac) [reply README.md:5]: https://stackoverflow.com/questions/20303826/how-to-highlight-bash-shell-commands-in-markdown

**Review by obiwac — APPROVED** [review]

**Comment by obiwac** [comment]
LGTM, I still haven't had the opportunity to test it, but yeah lgtm i'll merge it when i do

**Comment by RappyTV** [comment]
when are you going to test it? xd

**Comment by Emilevillette** [comment]
I will test it this week.

**Review by Emilevillette — APPROVED** [review]
Ok, nothing to report.

**Pull request merged**